src/commands/force/user/password/generate.ts

```diff
@@ -45,13 +69,44 @@ export class UserPasswordGenerateCommand extends SfdxCommand {
         const connection: Connection = await Connection.create({ authInfo });
         const org = await Org.create({ connection });
         const user: User = await User.create({ org });
-        const password = User.generatePasswordUtf8();
+        let passwordCondition: PasswordConditions;
+        if (connection.metadata) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const list: any = await connection.metadata.list({ type: 'ProfilePasswordPolicy' });
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-explicit-any
+          const profPasswordPolicies: ProfilePasswordPolicy = JSON.parse(
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+            JSON.stringify(await connection.metadata.read('ProfilePasswordPolicy', [list.fullName]))
+          );
+          passwordCondition = {
+            length: profPasswordPolicies.minimumPasswordLength,
+            complexity: profPasswordPolicies.passwordComplexity,
+          };
+
+          if (!passwordCondition.length || !passwordCondition.complexity) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+            const orgPasswordPolicy: OrgPasswordPolicy = JSON.parse(
+              JSON.stringify(await connection.metadata.read('SecuritySettings', ['passwordPolicies']))
+            );
+            passwordCondition.length = passwordCondition.length
+              ? passwordCondition.length
+              : orgPasswordPolicy.passwordPolicies?.minimumPasswordLength;
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+            passwordCondition.complexity = passwordCondition.complexity
+              ? passwordCondition.length
+              : PasswordComplexityTypes[orgPasswordPolicy.passwordPolicies.complexity];
+          }
+        } else
+          passwordCondition = {
+            length: 10,
+            complexity: 1,
+          };
+        const password = User.generatePasswordUtf8(passwordCondition);
         // we only need the Id, so instead of User.retrieve we'll just query
```

Comment:
you can try to get the user's profile, but have some logic to handle if we can't get it, or we can't get the policies (ex: user doesn't have metadata access)

Reply:
Currently i am checking if user.metadata is defined or not. If not I am generating the password with the default settings.
How to know whether the user have access or how we might be getting the errors ?

Reply:
There's a way to check the permissions, but it takes a call anyway , so I'd suggest a try-catch (where the catch would use the defaults because we can't come up with anything better).

That would also handle any other permissions weirdness we're not expecting.